test/powershell/Modules/Microsoft.PowerShell.Utility/XMLCommand.Tests.ps1

```diff
@@ -340,4 +346,200 @@ Describe "XmlCommand DRT basic functionality Tests" -Tags "CI" {
 		$cmd.Xml = $xml
 		$cmd.Xml | Should -Be $xml
 	}
+
+    Context "ConvertTo-CliXml & ConvertFrom-CliXml" {
+
+        It "Getting cmdlet info should work" {
+            $content = Get-Command export* -Type Cmdlet | Select-Object -First 3 | ConvertTo-CliXml
+            $results = ConvertFrom-CliXml $content
+            $results.Count | Should -Be 3
+            $results[0].PSTypeNames[0] | Should -BeExactly "Deserialized.System.Management.Automation.CmdletInfo"
+            $results[1].PSTypeNames[0] | Should -BeExactly "Deserialized.System.Management.Automation.CmdletInfo"
+            $results[2].PSTypeNames[0] | Should -BeExactly "Deserialized.System.Management.Automation.CmdletInfo"
+        }
+
+        It "Rehydration should work" {
+            $property1 = 256
+            $property2 = "abcdef"
+            $isHiddenTestType = [IsHiddenTestType]::New($property1,$property2)
+            $content = $isHiddenTestType | ConvertTo-CliXml
+            $results = ConvertFrom-CliXml $content
+            $results.Property1 | Should -Be $property1
+            $results.Property2 | Should -BeExactly $property2
+        }
+
+        It "ConvertTo-CliXml StopProcessing should succeed" {
+            $ps = [PowerShell]::Create()
+            $null = $ps.AddScript("1..10")
+            $null = $ps.AddCommand("foreach-object")
+            $null = $ps.AddParameter("Process", { $_; Start-Sleep -Seconds 1 })
+            $null = $ps.AddCommand("ConvertTo-CliXml")
+
+            Wait-UntilTrue { $ps.BeginInvoke() } -IntervalInMilliseconds 1000
+            $null = $ps.Stop()
+            $ps.InvocationStateInfo.State | Should -BeExactly "Stopped"
+            $ps.Dispose()
+        }
+
+        It "ConvertFrom-CliXml StopProcessing should succeed" {
+            $content = 1,2,3 | ConvertTo-CliXml
+            $ps = [PowerShell]::Create()
+            $ps.AddCommand("Get-Process")
+            $ps.AddCommand("ConvertFrom-CliXml")
+            $ps.AddParameter("InputObject", $content)
+            $ps.BeginInvoke()
+            $ps.Stop()
+            $ps.InvocationStateInfo.State | Should -BeExactly "Stopped"
+        }
+
+        It "Should serialize integers correctly using ValueFromPipeline" {
+            $testObject = 1,2,3
+            $content = $testObject | ConvertTo-CliXml
+            $testObject | Export-CliXml -Path $testfile
+            $content | Should -Be (Get-Content -Path $testfile -Raw)
```

Comment:
Maybe https://pester.dev/docs/v4/commands/Should#filecontentmatchexactly

Reply:
I think we can actually use https://pester.dev/docs/v4/commands/Should#filecontentmatchmultiline, which is better for matching multi line files.

Reply:
Fixed in fdb200380bc9d4c3bdf2da20e98c3e3bfe90617a.